ikarus/finiteelements/mechanics/truss.hh

```diff
@@ -0,0 +1,246 @@
+    const auto& d = par.globalSolution();
+    auto disp     = Ikarus::FEHelper::localSolutionBlockVector<Traits>(d, underlying().localView(), dx);
+    return disp;
+  }
+
+  /**
+   * \brief Gets the strain for the given Requirement and optional displacement vector.
+   *
+   * \tparam ST The scalar type for the strain.
+   * \param par The Requirement object.
+   * \param dx Optional displacement vector.
+   * \return A tuple containing Green-Lagrange strain, its first and second derivatives w.r.t displacements and
+   * the length of the undeformed and deformed geometry.
+   */
+  template <class ST = double>
+  KinematicVariables<ST> computeStrain(
+      const Requirement& par,
+      const std::optional<std::reference_wrapper<const Eigen::VectorX<ST>>>& dx = std::nullopt) const {
+    KinematicVariables<ST> kin;
+    const auto& localView    = underlying().localView();
+    const auto& tree         = localView.tree();
+    const auto numberOfNodes = tree.child(0).finiteElement().size();
+    auto& ele                = localView.element();
+    const auto X1            = Dune::toEigen(ele.geometry().corner(0));
+    const auto X2            = Dune::toEigen(ele.geometry().corner(1));
+    const auto u             = Dune::viewAsEigenMatrixAsDynFixed(displacement(par, dx)).transpose().eval();
+    const auto A1            = X2 - X1;
+    const auto ud1           = u.col(1) - u.col(0);
+
+    const Eigen::Vector<ST, worldDim> x1 = X1 + u.col(0);
+    const Eigen::Vector<ST, worldDim> x2 = X2 + u.col(1);
+
+    const double Lsquared = A1.squaredNorm();
+    const ST lsquared     = (x2 - x1).squaredNorm();
+
+    kin.L = sqrt(Lsquared);
+    kin.l = sqrt(lsquared);
+
+    // Linear strain
+    kin.Elin = (kin.l - kin.L) / kin.L;
+
+    // Green-Lagrange strains
+    kin.Egl = 0.5 * (lsquared - Lsquared) / Lsquared;
+
+    // First derivative of Egl w.r.t displacements
+    kin.dEdu.setZero(worldDim * numberOfNodes);
+    kin.dEdu << -A1 - ud1, A1 + ud1;
+    kin.dEdu /= Lsquared;
+
+    // Second derivative of Egl w.r.t displacements
+    kin.ddEddu.setIdentity(worldDim * numberOfNodes, worldDim * numberOfNodes);
+    kin.ddEddu.template topRightCorner<worldDim, worldDim>()   = -Eigen::Matrix<ST, worldDim, worldDim>::Identity();
+    kin.ddEddu.template bottomLeftCorner<worldDim, worldDim>() = -Eigen::Matrix<ST, worldDim, worldDim>::Identity();
+    kin.ddEddu /= Lsquared;
+
+    return kin;
+  }
+
+  using SupportedResultTypes =
+      std::tuple<decltype(makeRT<ResultTypes::cauchyAxialForce>()), decltype(makeRT<ResultTypes::PK2AxialForce>()),
+                 decltype(makeRT<ResultTypes::linearAxialForce>())>;
+
+private:
+  template <template <typename, int, int> class RT>
+  static consteval bool canProvideResultType() {
+    return isSameResultType<RT, ResultTypes::cauchyAxialForce> or isSameResultType<RT, ResultTypes::PK2AxialForce> or
+           isSameResultType<RT, ResultTypes::linearAxialForce>;
+  }
```

Comment:
You can do this in a more gerneral way like 
```cpp
  template <template <typename, int, int> class RT>
  static consteval bool canProvideResultType() {
    bool result;
    Dune::Hybrid::forEach(SupportedResultTypes(), [&]<typename RT_>(RT_ rt) {
      if constexpr (isSameResultType<RT, RT_::template Rebind>)
        result = true;
    });
    return result;
  }
  ```
  I will make a seperate PR where i generalize this functionality and make it such that every element only has to implement the tuple and not this function

Reply:
Yes basically as a base object maybe.

Reply:
Henri as you looking into this in a separate PR, I am not doing this step now.